File: 项目文件/backend/app/schemas/link_relation.py
"""关联关系 Pydantic 模型"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LinkRelationCreate(BaseModel):
    """创建关联关系请求"""

    source_type: str
    source_id: uuid.UUID
    target_type: str
    target_id: uuid.UUID


class LinkRelationResponse(BaseModel):
    """关联关系响应"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    source_type: str
    source_id: uuid.UUID
    target_type: str
    target_id: uuid.UUID
    created_at: datetime


class LinkRelationListResponse(BaseModel):
    """关联关系列表响应"""

    items: list[LinkRelationResponse]
    total: int

File: 项目文件/frontend/src/types/link-relation.ts
/** 通用关联关系类型定义 */

export interface LinkRelation {
  id: string;
  source_type: string;
  source_id: string;
  target_type: string;
  target_id: string;
  created_at: string;
}

export interface LinkRelationCreate {
  source_type: string;
  source_id: string;
  target_type: string;
  target_id: string;
}

export interface LinkRelationListResponse {
  items: LinkRelation[];
  total: number;
}

/** 关联实体（getLinkedEntities 返回项） */
export interface LinkedEntity {
  type: string;
  id: string;
}
